sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/display/DisplayDataTest.java

```diff
@@ -968,23 +965,48 @@ public void populateDisplayData(Builder builder) {
   }
 
   /**
-   * Validate that all runners are resilient to exceptions thrown while retrieving display data.
+   * Verify that {@link DisplayData.Builder} can recover from exceptions thrown in user code.
    */
   @Test
-  @Category(RunnableOnService.class)
-  public void testRunnersResilientToDisplayDataExceptions() {
-    Pipeline p = TestPipeline.create();
-    PCollection<Integer> pCol = p
-        .apply(Create.of(1, 2, 3))
-        .apply(new IdentityTransform<Integer>() {
-          @Override
-          public void populateDisplayData(Builder builder) {
-            throw new RuntimeException("bug!");
-          }
-        });
+  public void testCanRecoverFromBuildException() {
```

Comment:
Maybe extend the comment?

```
* Verify that {@link DisplayData.Builder} can recover from exceptions thrown in user code.
* This is not used within the Beam SDK since we want all code to produce valid DisplayData.
* This test just ensures it is possible to write custom code that does recover.
```